tests/func/test_data_cloud.py

```diff
@@ -570,3 +571,18 @@ def test_target_remote(tmp_dir, dvc, make_remote):
         "6b18131dc289fd37006705affe961ef8.dir",
         "b8a9f715dbb64fd5c56e7783c6820a61",
     }
+
+
+def test_push_pull_glob_no_match(
+    tmp_dir, dvc, erepo_dir, run_copy, local_remote
+):
+    tmp_dir.gen("foo", "foo")
+    erepo_dir.add_remote(config=local_remote.config)
+
+    with erepo_dir.chdir():
+        with pytest.raises(GlobDoesNotMatchError):
+            dvc.push(targets="b*", glob=True)
+
+        with pytest.raises(GlobDoesNotMatchError):
+            dvc.pull(targets="b*", glob=True)
+        assert not os.path.exists(erepo_dir.dvc.stage_cache.cache_dir)
```

Comment:
I think this is all we need for covering the changes:
 
```suggestion
def test_push_pull_glob_no_match(
    tmp_dir, dvc,
):
    with pytest.raises(GlobDoesNotMatchError):
        dvc.push(targets="b*", glob=True)

    with pytest.raises(GlobDoesNotMatchError):
        dvc.pull(targets="b*", glob=True)
```